glide-core/redis-rs/redis/src/cluster_async/connections_logic.rs

```diff
@@ -322,37 +325,67 @@ async fn create_and_setup_user_connection<C>(
     params: ClusterParams,
     socket_addr: Option<SocketAddr>,
     glide_connection_options: GlideConnectionOptions,
-) -> RedisResult<ConnectionWithIp<C>>
+) -> RedisResult<ConnectionDetails<C>>
 where
     C: ConnectionLike + Connect + Send + 'static,
 {
-    let mut connection: ConnectionWithIp<C> = create_connection(
+    let mut connection: ConnectionDetails<C> = create_connection(
         node,
         params.clone(),
         socket_addr,
         false,
         glide_connection_options,
     )
     .await?;
-    setup_user_connection(&mut connection.conn, params).await?;
+    setup_user_connection(&mut connection, params).await?;
     Ok(connection)
 }
 
-async fn setup_user_connection<C>(conn: &mut C, params: ClusterParams) -> RedisResult<()>
+async fn setup_user_connection<C>(
+    conn_details: &mut ConnectionDetails<C>,
+    params: ClusterParams,
+) -> RedisResult<()>
 where
     C: ConnectionLike + Connect + Send + 'static,
 {
+    // let mut conn = &conn_details.conn;
     let read_from_replicas = params.read_from_replicas
         != crate::cluster_slotmap::ReadFromReplicaStrategy::AlwaysFromPrimary;
     let connection_timeout = params.connection_timeout;
-    check_connection(conn, connection_timeout).await?;
+    check_connection(&mut conn_details.conn, connection_timeout).await?;
     if read_from_replicas {
         // If READONLY is sent to primary nodes, it will have no effect
-        crate::cmd("READONLY").query_async(conn).await?;
+        crate::cmd("READONLY")
+            .query_async(&mut conn_details.conn)
+            .await?;
     }
+
+    update_az_from_info(conn_details).await?;
     Ok(())
 }
 
+// Helper function to extract and update availability zone from INFO command
+async fn update_az_from_info<C>(conn_details: &mut ConnectionDetails<C>) -> RedisResult<()>
+where
+    C: ConnectionLike + Send + 'static,
+{
+    let info_res = conn_details.conn.req_packed_command(&cmd("INFO")).await;
```

Comment:
`INFO` is a slow and expensive command. We should check whether AZ awareness is required **before** we call `INFO`